draft-demarco-oauth-status-assertions.md

```diff
@@ -676,6 +680,123 @@ may not always coincide with the actual usability of a Digital Credential,
 allowing Verifiers to examine and make educated conclusions based on a
 variety of scenarios.
 
+# Detailed Status Assertions
+
+Status Assertions can introduce a more accurate level of detail, and therefore not necessarly limited to simple boolean information.
+This enables Verifier policies to be conditioned on the presence of secured information, instead of the absence of information.
+This section proposes syntax to support detailed assertions.
+The `credential_status_validity` claim MUST be present and be either `true` or `false`.
```

Comment:
Pending beeing neither valid or invalid...

Reply:
pending is an half state, my pythonic mind says that 0.5 is not true

![immagine](https://github.com/user-attachments/assets/d5bacd18-7e4b-4511-8615-c4d9be462d34)

Reply:
In the W3C Bitstring Status list, there is a purpose called "message" which is not revoked or suspended... You can use it to signal any message the issuer chooses.... Including pending.

The more I have thought about this, the more I conclude that credential status is a separate category of validity information... It's not as simple as being inside or outside a validity period... Or revoked or suspended being invalid and their opposites being valid.

I think it's better to allow the verifier to determine validity, and for the issuer to simply focus on clearly communicating.

Reply:
so why don't we elliminate the boolean entirely and just have an IANA registry for the status semantic - the spec would register "valid" and "invalid" as two standard semantics. Everybody else get to define their own. The problem with mixing a boolean with "extra information" is that you get into a criciality issue: what if somebody defines a semantic that critically modifies the truthyness of the boolean. This is something that has plagued extensions in X509 btw.

Reply:
I have come to the position that boolean values are a mistake for status assertions... and that it is verifier policy that determines if a status is cause to reject a presentation.

I've also come to the conclusion that IANA registries are not necessarily the best scalable solution for status assertions in complex domains (chemistry, logistics, healthcare, cyber).

If there are assertions that need to be "globally understood" then IANA makes sense, but I don't see a global registry that starts with "suspended" or "revoked" growing well.

I think many of the entries that might be entered in such a registry are probably much more domain specific.

The guidance to experts would typically be, "make sure there is no existing registry entry that solves this problem"... I imagine adjudicating new requests for all use cases in a single registry to be extremely difficult... and producing the same outcome you mention above, just with multiple status options, or multiple status values... criticality is lost, and verifier behavior becomes undefined.

In my mental model, status assertions are always credential type specific.

Although it might look like both a business license and a drivers license are "revoked"... "revoked" actually means different things, depending on the credential type.

We want to ensure that "revoked" does not mean different things for the same credential type across different issuers.

This is better accomplished with credential type registries, than status registries.

In a credential type registry, you could have:

+------------------+------------------------+
|  type                       |  assertions                       |
+------------------+------------------------+
|  drivers license      |  suspended, revoked      |
+------------------+------------------------+
|  business license   |  pending, revoked           |
+------------------+------------------------+

Statuses are relative to a credential type, and complexity and interoperability are constrained by "registered credential types"... In order to understand "drivers license", you MUST understand "suspended" or "revoked".

This approach also works in the open world model. 

For example, we use JSON Schema to enable this kind of status scheme for W3C Verifiable Credentials in a supply chain context.

The issuer can publish the documentation that verifiers can use to determine how to process the credential, and the status assertions for each credential type that the issuer supports.

If the credential contains a reference to its own schema, there is a single point point of discovery for all this information.

Coming back to the spec text we have.

I think it is fine for the issuer to try to be helpful and tell the verifier "new information, don't reject this", or "new information, please reject this".

But its still going to be the verifier's decision what to do... In that sense, I don't see the value of the boolean as being particularly high.

If the credential type tells you there must be a status assertion and the verifier does not get one, reject.
Otherwise, let the verifier apply their policy to the credential and the status assertion, and determine if they should reject or accept.

Reply:
In StatusList we have come to the same conclusion as Leif and we already have a regristration for the IANA Status semantic, you can think of piggybacking on this one. This is a potential for symmtry and symbioses between the specs.

Reply:
That is not a bad idea

Reply:
I'm very supportive of aligning the status list and status assertion claims and semantics everywhere we can.

Reply:
I filed https://github.com/peppelinux/draft-demarco-oauth-status-assertions/issues/81

Reply:
I support aligning the specs terms, since status assertions was born as a status list feature/extension

Reply:
@paulbastian could you please provide reference for the StatusList IANA Status semantics registration.

Reply:
I refer to this section: https://drafts.oauth.net/draft-ietf-oauth-status-list/draft-ietf-oauth-status-list.html#name-status-types
I just observed though that we are missing a corresponding subsection in the IANA as appendix, which I will fix shortly.